web_widget_text_markdown/static/src/lib/bootstrap-markdown.js

```diff
@@ -0,0 +1,1575 @@
+    },
+    getBulletNumber: function(startIndex) {
+      var bulletNum = this.getContent().slice(startIndex).split('.')[0];
+      return $.isNumeric(bulletNum) ? parseInt(bulletNum) : null;
+    },
+    change: function(e) {
+      this.$options.onChange(this);
+      return this;
+    },
+    select: function(e) {
+      this.$options.onSelect(this);
+      return this;
+    },
+    focus: function(e) {
+      var options = this.$options,
+        isHideable = options.hideable,
+        editor = this.$editor;
+
+      editor.addClass('active');
+
+      // Blur other markdown(s)
+      $(document).find('.md-editor').each(function() {
+        if ($(this).attr('id') !== editor.attr('id')) {
+          var attachedMarkdown;
+
+          if (attachedMarkdown = $(this).find('textarea').data('markdown'),
+            attachedMarkdown === null) {
+            attachedMarkdown = $(this).find('div[data-provider="markdown-preview"]').data('markdown');
+          }
+
+          if (attachedMarkdown) {
+            attachedMarkdown.blur();
+          }
+        }
+      });
+
+      // Trigger the onFocus hook
+      options.onFocus(this);
+
+      return this;
+    },
+    blur: function(e) {
+      var options = this.$options,
+        isHideable = options.hideable,
+        editor = this.$editor,
+        editable = this.$editable;
+
+      if (editor.hasClass('active') || this.$element.parent().length === 0) {
+        editor.removeClass('active');
+
+        if (isHideable) {
+          // Check for editable elements
+          if (editable.el !== null) {
+            // Build the original element
+            var oldElement = $('<' + editable.type + '/>'),
+              content = this.getContent(),
+              currentContent = this.parseContent(content);
+
+            $(editable.attrKeys).each(function(k, v) {
+              oldElement.attr(editable.attrKeys[k], editable.attrValues[k]);
+            });
+
+            // Get the editor content
+            oldElement.html(currentContent);
+
+            editor.replaceWith(oldElement);
+          } else {
+            editor.hide();
+          }
+        }
+
+        // Trigger the onBlur hook
+        options.onBlur(this);
+      }
+
+      return this;
+    }
+
+  };
+
+  /* MARKDOWN PLUGIN DEFINITION
+   * ========================== */
+
+  var old = $.fn.markdown;
+
+  $.fn.markdown = function(option) {
+    return this.each(function() {
+      var $this = $(this),
+        data = $this.data('markdown'),
+        options = typeof option == 'object' && option;
+      if (!data)
+        $this.data('markdown', (data = new Markdown(this, options)));
+    });
+  };
+
+  $.fn.markdown.messages = {};
+
+  $.fn.markdown.defaults = {
+    /* Editor Properties */
+    autofocus: false,
+    hideable: false,
+    savable: false,
+    width: 'inherit',
+    height: 'inherit',
+    resize: 'none',
+    iconlibrary: 'glyph',
```

Comment:
Shouldn't it be `fa`? Otherwise the icons aren't shown if glyphicon isn't available (and usually it isn't)